src/functions-and-arrays.js

```diff
@@ -1,38 +1,106 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers() {
+  if (arguments[0] < arguments[1]) {
+    return arguments[1];
+  }
+  if (arguments[0] > arguments[1]) {
+    return arguments[0];
+  }
+  else {
+    return arguments[0]
+  }
+}
 
+console.log(maxOfTwoNumbers(1, 2))
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
-
-function findLongestWord() {}
-
+//const wordstest =[]
+
+  function findLongestWord(words) {
+    let longestWord = ""
+    if (words.length === 0) {
+      return longestWord = null
+    }
+    for (let i=0; i<words.length; i++) {
+      if (longestWord.length < words[i].length) {
+        longestWord = words[i]
+      }
+    }
+    return longestWord
+    }
+// note to me: das null if am Anfang, weil wenn ich das in den loop packe, dann kommt er nie zu der condition
+  
+console.log(findLongestWord(words))
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
+function sumNumbers(numbers) {
+  let addUp = 0;
+  for (number of numbers) {
+    addUp += number;
+  }
+  return addUp;
+}
 
+console.log(sumNumbers(numbers))
 
 // Iteration #3.1 Bonus:
-function sum() {}
-
+const mixedArr = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10];
+function sum(mixedArr) {
```

Comment:
Missing throwing an error an Unsuported type of data. ie: Object or Array, but both typeof come as "object"